godel_surface_detection/src/services/blending_service_path_generation.cpp

```diff
@@ -384,122 +147,188 @@ SurfaceBlendingService::generateProcessPath(const int& id,
+    geometry_msgs::Pose geo_pose;
+    std::vector<Eigen::Matrix4d, Eigen::aligned_allocator<Eigen::Matrix4d>> poses;
 
-    if(requestEdgePath(sorted_boundaries, i, SS, edge_visualization, edge_poses))
-    {
-      ProcessPathResult::value_type edge_path_result; // pair<string, geometry_msgs::PoseArray>
-      edge_path_result.first = name + "_edge_" + std::to_string(i);
-
-      /*
-       * Set the orientation for all edge points to be the orientation of the surface normal
-       * This is a hack that should be removed when planar surfaces assumption is dropped
-       * The main purpose here is to "smooth" the trajectory of the edges w.r.t. the z axis
-      */
-      for(auto& p : edge_poses.poses)
-        p.orientation = boundary_pose.orientation;
-
-      edge_path_result.second = edge_poses;
-      result.paths.push_back(edge_path_result);
-
-      // Add poses to visualization
-      all_edge_poses.poses.insert(std::end(all_edge_poses.poses), std::begin(edge_poses.poses),
-                                  std::end(edge_poses.poses));
-
-      // Add edge to data coordinator
-      data_coordinator_.addEdge(id, edge_path_result.first, edge_poses);
-    }
-    else
+    // Get boundary trajectory and trim last two poses (last poses are susceptible to large velocity changes)
+    SS.getBoundaryTrajectory(sorted_boundaries, i, poses);
+    poses.resize(poses.size() - 2);
+
+    // Convert eigen poses to geometry poses for messaging and visualization
+    for(const auto& p : poses)
     {
-      // Blend path request failed
-      ROS_WARN_STREAM("Could not calculate blend path for surface: " << name);
+      Eigen::Affine3d pose(p.matrix());
+      tf::poseEigenToMsg(pose, geo_pose);
+      edge_poses.poses.push_back(geo_pose);
     }
+
+    result.push_back(edge_poses);
+  }
+  return result.size() > 0;
+}
+
+
+bool
+SurfaceBlendingService::generateProcessPath(const int& id,
+                                            ProcessPathResult& result)
+{
+  using godel_surface_detection::detection::CloudRGB;
+
+  std::string name;
+  pcl::PolygonMesh mesh;
+  CloudRGB::Ptr surface_ptr (new CloudRGB);
+
+  data_coordinator_.getSurfaceName(id, name);
+  data_coordinator_.getSurfaceMesh(id, mesh);
+  data_coordinator_.getCloud(godel_surface_detection::data::CloudTypes::surface_cloud, id, *surface_ptr);
+  return generateProcessPath(id, name, mesh, surface_ptr, result);
+}
+
+
+bool
+SurfaceBlendingService::generateProcessPath(const int& id,
+                                            const std::string& name,
+                                            const pcl::PolygonMesh& mesh,
+                                            godel_surface_detection::detection::CloudRGB::Ptr surface,
+                                            ProcessPathResult& result)
+{
+  std::vector<geometry_msgs::PoseArray> blend_result, edge_result, scan_result;
+
+  pluginlib::ClassLoader<path_planning_base::PathPlanningBase> poly_loader("path_planning_plugins",
```

Comment:
Would you consider making the `ClassLoader` objects themselves a part of the SurfaceBlendingService?